custom/onse/tasks.py

```diff
@@ -0,0 +1,298 @@
+    # TB COMM Number of Functional Sputum sample collection points in the catchment (GhUn5j5ajzK) not found
+    # NCD CC HIV Status +Ve on ART (E2TaryAVqeT) not found
+    'dhis_village_clinics': DataElement(
+        id='gtLvoz94gur',  # CHD IMCI # of Functional Village Clinics Within Catchment
+        data_set='hWDsGIjs16g',
+    ),
+    'dhis_cbdas': DataElement(
+        id='LGaHPDsUydT',  # HTS Number of CBDA/HSA
+        data_set='wLQlOnKX6yN',
+    ),
+    'dhis_village_health_committees': DataElement(
+        id='J7fogdejE3j',  # HMIS HM Active village health committees within catchment area
+        data_set='q1Es3k3sZem',
+    ),
+    'dhis_households_with_improved_latrines': DataElement(
+        id='u5erlSYbxTU',  # ENVT EH # Of Households Owning And Using Improved Sanitary Facilities
+        data_set='iTo9FkAJTSl',
+    ),
+    'dhis_households_access_to_clean_water': DataElement(
+        id='BDzMvX3y7Kc',  # HMIS # of Households with Access to Safe Drinking Water
+        data_set='q1Es3k3sZem',
+    ),
+}
+
+DOMAIN = 'onse-iss'
+CASE_TYPE = 'facility_supervision'
+CONNECTION_SETTINGS_NAME = 'DHIS2 Facilities Import Server'
+
+
+_soft_assert = soft_assert('@'.join(('nhooper', 'dimagi.com')))
+
+
+@periodic_task(
+    # Run on the 5th day of every quarter
+    run_every=crontab(day_of_month=5, month_of_year='1,4,7,10',
+                      hour=22, minute=30),
+    queue='background_queue',
+)
+def update_facility_cases_from_dhis2_data_elements():
+    try:
+        dhis2_server = ConnectionSettings.objects.get(
+            domain=DOMAIN, name=CONNECTION_SETTINGS_NAME
+        )
+    except ConnectionSettings.DoesNotExist:
+        _soft_assert(False, (
+            f'ConnectionSettings {CONNECTION_SETTINGS_NAME!r} not found in '
+            f'domain {DOMAIN!r} for importing DHIS2 data elements.'))
+        return
+
+    facility_case_blocks = get_case_blocks()
+    facility_case_blocks = set_case_updates(dhis2_server, facility_case_blocks)
+    save_cases(facility_case_blocks)
+
+
+def get_case_blocks() -> Iterable[CaseBlock]:
+    case_accessors = CaseAccessors(DOMAIN)
+    for case_id in case_accessors.get_case_ids_in_domain(type=CASE_TYPE):
+        case = case_accessors.get_case(case_id)
+        if not case.external_id:
+            # This case is not mapped to a facility in DHIS2.
+            continue
+        case_block = CaseBlock(
+            case_id=case.case_id,
+            external_id=case.external_id,
+            case_type=CASE_TYPE,
+            case_name=case.name,
+            update={},
+        )
+        yield case_block
+
+
+def set_case_updates(
+    connection_settings: ConnectionSettings,
+    case_blocks: Iterable[CaseBlock]
+) -> Iterable[CaseBlock]:
+    """
+    Fetch all the data elements for the data set, because we don't know
+    in advance what category option combos to query for
+    """
+    # e.g. https://play.dhis2.org/dev/api/dataValueSets?orgUnit=jNb63DIHuwU
```

Comment:
would it be straightforward to use this example as a test case?  It'd be nice to declare what the expected response format is so we can adapt easily if it changes.

Reply:
I'm a little hesitant to write automated tests that hit play.dhis2.org, but I have included [a sample return value in a doctest](https://github.com/dimagi/commcare-hq/pull/28742/commits/4a6e866d77efec93128a0754db878c711840ac92#diff-bcabc4569dc5b1ece3959f73e371f4ebd74b1862361b35347da01b2a88313389R239-R254). Do you think that's enough?

Reply:
Oh yeah, that's basically what I meant - stubbing in an example response like what would be provided by the API.